datanode/sqlstore/teams.go

```diff
@@ -121,13 +121,13 @@ func (t *Teams) RefereeSwitchedTeam(ctx context.Context, referee *entities.Refer
 	return err
 }
 
-func (t *Teams) GetTeam(ctx context.Context, teamID entities.TeamID, partyID entities.PartyID) (entities.Team, error) {
+func (t *Teams) GetTeam(ctx context.Context, teamID entities.TeamID, partyID entities.PartyID) (*entities.Team, error) {
 	defer metrics.StartSQLQuery("Teams", "GetTeam")()
 
 	var team entities.Team
 
 	if teamID == "" && partyID == "" {
```

Comment:
Does this useful user-input error ever make its way out atm? I think current it'll end up being hitting the `return nil, formatE(ErrTeamNotFound, err)`. We should probably validate it earlier in `func (t *TradingDataServiceV2) ListTeams()` itself.

Reply:
So the GetTeam method on the service was because originally I had intended a GetTeam API for the gRPC, but later it just made more sense to add the Filter to the ListTeams API and use this method on the service to retrieve the team if the team ID or party ID filter is provided.

The ListTeams gRPC API can accept an empty Team ID and Party ID, in which case it will list all teams, so it doesn't make sense to have this check in the gRPC method itself.